src/jit/codegenlegacy.cpp

```diff
@@ -11371,8 +11371,6 @@ void CodeGen::genCodeForTreeSmpOpAsg(GenTreePtr tree)
 
             if (VarSetOps::IsMember(compiler, genUpdateLiveSetForward(op2), varDsc->lvVarIndex))
             {
-                noway_assert(compiler->opts.compDbgCode);
```

Comment:
If this assert fires, it indicates that the variable is live after encountering a def. I believe this assert should still be valid. You shouldn't delete it without understanding why it is firing.

---
In reply to: [78632294](https://github.com/dotnet/coreclr/pull/7012#discussion_r78632294) [](ancestors = 78632294)